### DISCUSSION TITLE
When a task fails in the body of a chord, and that chord chains into another task, there another buggy exception raised

### DISCUSSION THREAD

**Discussion by mscout1** [celery/celery#8127] [Q&A]
I have a chord. This chord is the first step of a chain. When one of the tasks in that chord fails, I get a `ChordError` as expected but then there is a `TypeError: sequence item 1: expected a bytes-like object, NoneType found` that replaces it. 

If I leave off the chain, and just run the chord, It does not have this second exception. In that case, I get the original exception from the chord head task when I wait. 

If I chain, and cause the second exception, *that* exception replaces my original and obscures the cause of the problem. 

I have a toy repro script. 

```python
from celery import Celery, group
app = Celery('repro_toy')
app.conf.update(
    broker_url='redis://localhost/0',
    result_backend='redis://localhost/0',
    #task_serializer='pickle',
    #result_serializer='pickle',
    #accept_content=['application/json', 'application/x-python-serialize']
    )

@app.task(bind=True)
def debug_A(self, i):
    print(f'Doing A{i}')

@app.task(bind=True)
def debug_broke(self):
    print(f'OH NO')
    raise Exception("Head Aspload")

@app.task(bind=True)
def debug_finisher(self):
    print(f'OK, We did it all')
    
def probe_results(res):
    print (f"waiting for {res=}")
    res.wait()
    print(f"{res.failed()=}")
    print(f"{res.get()=}")
    
    
def launch(): 
    # This finishes fine 
    g = [debug_A.si(i) for i in range(5)]
    r = group(g) | debug_finisher.si() | debug_A.si(-1)
    res = r.apply_async()
    probe_results(res)

def launch_fail_better(): 
    # This blows up with a nice expected celery.exceptions.ChordError which seems correct
    # When I get reults I get my "Head Aspload" Exception
    g = [debug_A.si(i) for i in range(5)]
    g.append(debug_broke.si())
    r = group(g) | debug_finisher.si()
    res = r.apply_async()
    probe_results(res)

def launch_fail_bad(): 
    # This blows up with TypeError: sequence item 1: expected a bytes-like object, NoneType found
    g = [debug_A.si(i) for i in range(5)]
    g.append(debug_broke.si())
    r = group(g) | debug_finisher.si() | debug_A.si(-1)
    res = r.apply_async()
    probe_results(res)
``` 

Is this a bug in celery or am I doing something wrong?

**Comment by auvipy** [comment]
which version of celery you are using? as you are facing ChordError it is probably an issue on you code

**Reply by mscout1** [reply]
I have Celery 5.2.7. 

The ChordError is not the problem. I want the ChordError because it brings me the original error. The problem is the TypeError that I get instead of the ChordError. 

And yes, there is an issue in my code. The premise of this is that one of my tasks is raising an exception. The bug is in how that error is handled by celery.

**Reply by auvipy** [reply]
can you try a failing test case in the celery test suite?
